src/utils/blob.h

```diff
@@ -109,7 +109,15 @@ class blob
     /// NOTE: this operation is not efficient since it involves a memory copy.
     [[nodiscard]] static blob create_from_bytes(const char *s, size_t len)
     {
-        CHECK_NOTNULL(s, "null source pointer would lead to undefined behaviour");
+        if (s == nullptr) {
```

Comment:
If this happens in rare cases, wrap it by dsn_unlikely.

Reply:
OK, I dropped this and use `DCHECK` instead.